### PR TITLE
Use a different  task_order property to key on unique task orders

### PR DESCRIPTION
**Pull request by johnclary**
As we transition to using S3 as the source for task order data, we will no longer have the unique Knack `id` column in the [Socrata task orders dataset](https://data.austintexas.gov/Transportation-and-Mobility/-TEST-Mobility-Project-Task-Orders/ggav-ufvc). We use the `id` value in Moped to key on a unique task order when adding/removing task orders to a project record. 

This PR swaps out `id` for the the `task_order.task_order` property, which is an alphanum string that uniquely identifies each task order. 

Until this is merged, it’s now possible to duplicate the same task order on the funding tab.
Screen Shot 2022-03-24 at 3 53 20 PM

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/7931

## Testing
**URL to test:** [Netlify dep prev](https://deploy-preview-607--atd-moped-main.netlify.app/)

**Steps to test:**
Create a signal project.
Add and remove multiple task orders to the project.
Check the signals dashboard and make sure task orders are rendered correctly there.

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [x] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/views/projects/projectView/ProjectFundingTable.js
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectFundingTable.js:146]: lintings ☝️

**Comment by johnclary** [comment]
@amenity this needs to ship to **production** before/at the same time as Charlie finishes up the [new ETLs](https://github.com/cityofaustin/atd-data-tech/issues/7931) - !

**Review by frankhereford — APPROVED** [review]
mission-complete-spongebob

Code looks great & works great. Put a bow on this one - it's 💯.

**Review by amenity — APPROVED** [review]
@johnclary - works for me. 

I believe we only use **production** as the base for release candidates and patch PRs. Is your concern that merging this into **main** will break task order functionality in Staging until Charlie's ETL PR is merged? Do both changes belong in the same PR? 

Just some rambling thoughts, though. I totally defer to @mateoclarke on this.

File: moped-editor/package.json
Comment (mateoclarke) [reply moped-editor/package.json:5]: @johnclary, is this change intended as a patch? Right now its PR'ed against `main` so I think the version number should stay "1.2.0" for the next release. Otherwise we should PR against `production` for a patch right?

**Comment by mateoclarke** [comment]
> @mateoclarke @amenity this needs to ship to **production** before/at the same time as Charlie finishes up the [new ETLs](https://github.com/cityofaustin/atd-data-tech/issues/7931) - !
> 
> should i make this PR against prod instead? assuming that's true, i will re-base. lmk

I'd say keep this PR but go ahead and make another PR for the same branch that compares to `production`. We can hold it with a WIP label until the ETL is done.

**Review by mateoclarke — APPROVED** [review]
👍

File: moped-editor/package.json
Comment (frankhereford) [reply moped-editor/package.json:5]: John and I got a chance to talk about this yesterday in the dev sync. He indicated that he will put together a PR to get this merged into production. We also spoke about how I had flipped this diff in my mind, thinking that he was bumping the main branch to 1.2.0, which was an oversight on my part.

@johnclary - Since you're out of town, I hope you don't mind if I push a commit to this branch that nullifies this change to maintain main at 1.2.0.

**Review by chiaberry — APPROVED** [review]

**Comment by johnclary** [comment]
@frankhereford @mateoclarke since my initial PR branched from main, i rebased this one onto prod. can y'all give this another quick review? this will be merged as a hotfix and bump us to `v1.1.1`.

**Review by chiaberry — APPROVED** [review]

**Review by mateoclarke — APPROVED** [review]
🚢 💨 ✨

**Review by frankhereford — APPROVED** [review]
This looks good to me. 🚢 🚢 🚢

**Pull request merged**
